packages/react/radio-group/src/Radio.tsx

```diff
@@ -155,6 +140,49 @@ RadioIndicator.displayName = INDICATOR_NAME;
 
 /* ---------------------------------------------------------------------------------------------- */
 
+type BubbleInputProps = Omit<React.ComponentProps<'input'>, 'checked'> & {
+  checked: boolean;
+  control: HTMLElement | null;
+  stoppedPropagation: boolean;
+};
+
+const BubbleInput = (props: BubbleInputProps) => {
```

Comment:
This is exactly the same as `Switch` and mostly the same as `Checkbox` which just handles `indeterminate` as well. I dunno if we wanna start thinking about a re-useable thing for this in a new PR?

Reply:
Yeah I think we should probably look at it next.